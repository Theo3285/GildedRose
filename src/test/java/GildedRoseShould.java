import org.approvaltests.Approvals;
import org.junit.Test;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Random;

public class GildedRoseShould {

    private static final int SEED = 100;
    private static final int MINIMUM = -50;
    private static final int MAXIMUM = 101;
    private static final int NUMBER_OF_RANDOM_ITEMS = 2000;

    private List<String> itemNames = Arrays.asList(
            "+5 Dexterity Vest",
            "Aged Brie",
            "Elixir of the Mongoose",
            "Sulfuras, Hand of Ragnaros",
            "Backstage passes to a TAFKAL80ETC concert",
            "Conjured Mana Cake");

    private Random random = new Random(SEED);

    @Test
    public void update_quality() {
        List<Item> items = new ArrayList<Item>();
        for (int i = 0; i < NUMBER_OF_RANDOM_ITEMS; i++) {
            items.add(new Item(itemName(), sellIn(), quality()));
        }

        GildedRose gildedRose = new GildedRose(items);

        gildedRose.updateQuality();

        Approvals.verify(buildApprovedOutputFrom(items));
    }

    private String itemName() {
        return itemNames.get(random.nextInt(itemNames.size()));
    }

    private int sellIn() {
        return generateRandomNumber();
    }

    private int quality() {
        return generateRandomNumber();
    }

    private int generateRandomNumber(){
        return MINIMUM + random.nextInt(MAXIMUM);
    }

    private String buildApprovedOutputFrom(List<Item> items) {
        StringBuilder builder = new StringBuilder();
        for (Item item : items) {
            builder.append(item.toString() + "\r");
        }
        return builder.toString();
    }
}