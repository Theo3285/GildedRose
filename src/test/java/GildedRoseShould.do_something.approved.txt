Item{name='+5 Dexterity Vest', sellIn=9, quality=19}Item{name='Aged Brie', sellIn=1, quality=1}Item{name='Elixir of the Mongoose', sellIn=4, quality=6}Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=14, quality=21}Item{name='Conjured Mana Cake', sellIn=2, quality=5}